DDOT_module/child_classes/noise_model_generator_vanilla.py

```diff
@@ -0,0 +1,539 @@
+                        new_lists[i] = anf.lists_sum(cl0, cl1)
+
+            unique_stuff = [sorted(lis) for lis in np.unique(new_lists)]
+            new_lists = copy.deepcopy(unique_stuff)
+
+        clusters_list = new_lists
+
+        # Chop clusters if they exceed max size
+        chopped_clusters = []
+        for cluster in clusters_list:
+            if len(cluster) > maximal_size:
+                correlations_sorting = []
+                for qi in cluster:
+                    # as figure of merit, we will sum all correlations that are between
+                    # given qubit and other guys in its cluster.
+                    x = 0.0
+                    for list_now in clusters['q%s' % qi]:
+                        x += np.max([list_now[1], list_now[2]])
+
+                    correlations_sorting.append([qi, x])
+
+                correlations_sorted = sorted(correlations_sorting, key=lambda arg: arg[1],
+                                             reverse=True)
+
+                # choose only "maximal_size" qubits to belong to given cluster
+                qubits_sorted = [correlations_sorted[index][0] for index in range(maximal_size)]
+            else:
+                qubits_sorted = cluster
+            chopped_clusters.append(qubits_sorted)
+
+        chopped_clusters_sorted = sorted(chopped_clusters, key=lambda y: y[0])
+
+        self._clusters_list = chopped_clusters_sorted
+
+        return chopped_clusters_sorted
+
+    def _find_neighbors_of_cluster_holistic(self,
+                                            cluster: List[int],
+                                            maximal_size: int,
+                                            chopping_threshold: Optional[float] = 0.) -> List[int]:
+        """
+        For potentially_stochastic_matrix given cluster of qubits, find qubits which are their neighbors, i.e., they affect the
+        noise matrix of cluster significantly. Figure of merit for correlations here is:
+
+        c_{j -> cluster} = 1/2 || Lambda_{cluster}^{Y_j='0'}- Lambda_{cluster}^{Y_j='1'}||_{l1}
+
+        where Lambda_{cluster}^{Y_j} is the noise matrix describing noise on qubits in "cluster"
+        provided that input state of qubit "j" was "Y_j".
+        See also description of self._compute_clusters_pairwise.
+
+
+        :param cluster: list of labels of qubits in potentially_stochastic_matrix cluster
+        :param maximal_size: maximal allowed size of the set "cluster+neighborhood"
+        :param chopping_threshold: numerical value, for which correlations lower than
+              chopping_threshold are set to 0.
+              If not provided, it adds all_neighbors until maximal_size is met.
+
+
+
+        :return: neighbors_list: list of lists, each representing potentially_stochastic_matrix single cluster
+        """
+
+        size_cut = maximal_size - len(cluster)
+
+        potential_neighbours = []
+        for qi in self._qubit_indices:
+            if qi not in cluster:
```

Comment:
You can reduce nesting here by:
```suggestion
            if qi in cluster:
                continue
```